substrate-node/pallets/pallet-dao/src/dao.rs

```diff
@@ -0,0 +1,366 @@
+        proposal_hash: T::Hash,
+        approve: bool,
+    ) -> DispatchResultWithPostInfo {
+        ensure!(
+            T::Tfgrid::is_farm_owner(farm_id, who.clone()),
+            Error::<T>::NotAuthorizedToVote
+        );
+
+        let stored_proposal =
+            <Proposals<T>>::get(proposal_hash).ok_or(Error::<T>::ProposalMissing)?;
+
+        let mut voting = Self::voting(proposal_hash).ok_or(Error::<T>::ProposalMissing)?;
+        ensure!(
+            voting.index == stored_proposal.index,
+            Error::<T>::WrongIndex
+        );
+
+        // Don't allow votes after the time limit reached
+        ensure!(
+            frame_system::Pallet::<T>::block_number() <= voting.end,
+            Error::<T>::TimeLimitReached
+        );
+
+        let position_yes = voting.ayes.iter().position(|a| a.farm_id == farm_id);
+        let position_no = voting.nays.iter().position(|a| a.farm_id == farm_id);
+
+        // Detects first vote of the member in the motion
+        let is_account_voting_first_time = position_yes.is_none() && position_no.is_none();
+
+        if approve {
+            if position_yes.is_none() {
+                voting.ayes.push(proposal::VoteWeight {
+                    farm_id: farm_id,
+                    weight: Self::get_vote_weight(farm_id)?,
+                });
+            } else {
+                return Err(Error::<T>::DuplicateVote.into());
+            }
+            if let Some(pos) = position_no {
+                voting.nays.swap_remove(pos);
+            }
+        } else {
+            if position_no.is_none() {
+                voting.nays.push(proposal::VoteWeight {
+                    farm_id: farm_id,
+                    weight: Self::get_vote_weight(farm_id)?,
+                });
+            } else {
+                return Err(Error::<T>::DuplicateVote.into());
+            }
+            if let Some(pos) = position_yes {
+                voting.ayes.swap_remove(pos);
+            }
+        }
+
+        let yes_votes = voting.ayes.len() as u32;
+        let no_votes = voting.nays.len() as u32;
+        Self::deposit_event(Event::Voted {
+            account: who,
+            proposal_hash,
+            voted: approve,
+            yes: yes_votes,
+            no: no_votes,
+        });
+
+        Voting::<T>::insert(&proposal_hash, voting);
+
+        if is_account_voting_first_time {
+            Ok(Pays::No.into())
+        } else {
+            Ok(Pays::Yes.into())
+        }
+    }
+    pub fn _veto(who: T::AccountId, proposal_hash: T::Hash) -> DispatchResultWithPostInfo {
+        Self::is_council_member(who.clone())?;
+
+        let stored_proposal =
+            <Proposals<T>>::get(proposal_hash).ok_or(Error::<T>::ProposalMissing)?;
+
+        let mut voting = Self::voting(proposal_hash).ok_or(Error::<T>::ProposalMissing)?;
+        ensure!(
+            voting.index == stored_proposal.index,
+            Error::<T>::WrongIndex
+        );
+
+        // push vote to vetos
```

Comment:
Comment is not needed in my opinion.

Reply:
done in 4c5945b7c0b08acb7eb1c641fd20d8f32df7559c